fpnge.cc

```diff
@@ -638,8 +644,9 @@ void TryPredictor(size_t bytes_per_line_buf, const unsigned char *mask,
 }
 
 // Either bits_hi is empty, or bits_lo contains exactly mid_lo_nbits bits.
-FORCE_INLINE void WriteBits(MIVEC nbits, MIVEC bits_lo, MIVEC bits_hi,
-                            size_t mid_lo_nbits, BitWriter *writer) {
+static FORCE_INLINE void WriteBits(MIVEC nbits, MIVEC bits_lo, MIVEC bits_hi,
+                                   size_t mid_lo_nbits,
+                                   BitWriter *__restrict writer) {
```

Comment:
Can we make __restrict a macro?

Reply:
What sort of macro?

Your original code only supported GCC-compatible compilers. I added support for MSVC. Both of these recognise `__restrict`.

I generally use the [Hedley library](https://nemequ.github.io/hedley/) for dealing with compiler specific functionality, but I don't know whether you'd be interested in it. If so, it can be added in a future change.

Reply:
I mean just a #define RESTRICT __restrict - if someone someday wants to add support for some other compiler, they just need to change that line :)

Reply:
What other compiler did you have in mind? GCC-compatible and MSVC is like 99.99999% of usage out there.

If you're imagining some hypothetical compiler >10 years in the future, there's little reason for it to not adopt what all existing compilers support.  
And even if we get super lucky here, a find+replace is hardly more difficult for this future person to do.

Reply:
I think the main one I was thinking of is ICC, although it's unclear to me whether it supports __restrict or not...

I'm fine with leaving it as is for now.

Reply:
I see.  
ICC is GCC-compatible (or at least aims to be) and [supports `__restrict` fine](https://godbolt.org/z/nEf78d98r) (it's also superseded by ICX, so less relevant into the future).